.github/workflows/rust.yml

```diff
@@ -66,60 +57,106 @@ jobs:
       uses: arduino/setup-protoc@v1
       with:
         repo-token: ${{ secrets.GITHUB_TOKEN }}
-    - name: Set environment variables
-      run: |
-        echo "LINERA_STORAGE_SERVICE=127.0.0.1:1235" >> "$GITHUB_ENV"
-        echo "LINERA_WALLET=/tmp/local-linera-net/wallet_0.json" >> "$GITHUB_ENV"
-        echo "LINERA_STORAGE=rocksdb:/tmp/local-linera-net/client_0.db" >> "$GITHUB_ENV"
-        echo "LINERA_FAUCET_URL=http://localhost:8079" >> "$GITHUB_ENV"
-    - name: Build example applications
-      run: |
-        cd examples
-        cargo build --locked --release --target wasm32-unknown-unknown
-    - name: Run the storage-service instance and the storage-service tests
+    - name: Run the storage-service instance
       run: |
         cargo run --release -p linera-storage-service -- memory --endpoint $LINERA_STORAGE_SERVICE &
-        cargo test --features storage-service,unstable-oracles -- storage_service --nocapture
-    - name: Run Ethereum tests
-      run: |
-        cargo test -p linera-ethereum --features ethereum
-        cargo test test_wasm_end_to_end_ethereum_tracker --features ethereum,storage-service,unstable-oracles
-    - name: Run the benchmark test
-      run: |
-        cargo build --locked -p linera-service --bin linera-benchmark --features benchmark,storage-service
-        cargo test --locked -p linera-service --features benchmark,storage-service benchmark
     - name: Run the validators
       run: |
         cargo build --features storage-service
         mkdir /tmp/local-linera-net
         cargo run --features storage-service --bin linera -- net up --storage service:tcp:localhost:1235:table --policy-config devnet --path /tmp/local-linera-net --validators 4 --shards 4 &
-    - name: Compile Wasm test modules for Witty integration tests
-      run: |
-        cargo build -p linera-witty-test-modules --target wasm32-unknown-unknown
     - name: Run the faucet
       run: |
         cargo build --bin linera
         cargo run --bin linera -- faucet --amount 1000 --port 8079 &
-    - name: Check that the WIT files are up-to-date
-      run: |
-        cargo run --bin wit-generator -- -c
     - name: Run the remote-net tests
       run: |
         cargo test -p linera-service remote_net_grpc --features remote-net
-    - name: Run all tests using the default features (plus unstable-oracles, except storage-service)
+
+  execution-wasmtime-test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions-rust-lang/setup-rust-toolchain@v1
+    - name: Clear up some space
       run: |
-        # TODO(#2764): Actually link this to the default features
-        cargo test --no-default-features --features fs,macros,wasmer,rocksdb,unstable-oracles --locked
+        sudo rm -rf /usr/share/dotnet
+        sudo rm -rf /opt/ghc
+        sudo rm -rf "/usr/local/share/boost"
+        sudo rm -rf "$AGENT_TOOLSDIRECTORY"
```

Comment:
This runs in parallel to the `remote-net-tests` and also has the `"Clear up some space"` step. Are you sure these parallel jobs, trying to access the same directories, won't conflict with each other? Do they run on separate machines?